src/MergeContext.js

```diff
@@ -173,18 +263,20 @@ class MergeContext {
         // cannot be 'diverged' because _needRestart() succeeded
         assert(compareStatus === "ahead");
 
-        const commitStatus = await this._checkStatuses(this._tagSha, Config.stagingChecks(), true);
-        if (commitStatus === 'pending') {
-            this._log("waiting for more staging checks completing");
-            return StepResult.Suspend();
-        } else if (commitStatus === 'failure') {
+        const commitStatus = await this.__checkStagingStatuses();
+        this._log("status details: " + commitStatus);
+        if (commitStatus.failed()) {
             this._log("staging checks failed");
             return await this._cleanupMergeFailed(false, this._labelFailedStagingChecks);
+        } else if (commitStatus.pending()) {
+            this._log("waiting for more staging checks completing");
+            return StepResult.Suspend();
+        } else {
+            assert(commitStatus.succeeded());
+            await this._supplyStagingWithPrRequired(commitStatus);
+            this._log("staging checks succeeded");
```

Comment:
@rousskov, your [algorithm](https://measurementfactory.atlassian.net/browse/SQUID-330?focusedCommentId=39140&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-39140) for staged commits is implemented here: (1) at 268, (2) and (3) at 271, extra step (Copy all required recorded PR checks...) at 276.